### ISSUE TITLE
in terminal or SDL mode isn't possible copy|paste text on text screen

### ISSUE THREAD

**Issue by fhanzlik** [stsp/dosemu2#361]
In dosemu 1.4 I used very frequently to copy and paste text fragments in my DOS accounting program. It was possible copy and paste text from DOS to DOS, from DOS to Linux and from Linux to DOS (using primary selection/clipboard - select text and copy it with middle mouse button).
Now I just tried dosemu2 devel version  444.gf14c6fc (on Fedora 19/i686), and it seems as it isn't working generally:
- in terminal mode (dosemu -t) mouse cursor has the shape of arrows and isn't possible to select text, text selected in Linux isn't possible to paste to DOS.

- in SDL mode (dosemu -S) mouse cursor isn't visible in DOS window and isn't possible select text there, also text selected in Linux isn't possible to paste to DOS.

- in X mode (dosemu -X) all work as expected, select/paste DOS->DOS, DOS->Linux, Linux->DOS.

Not sure when is intended to be functional also clipboard copy/paste (copy selected text into it with CTRL/C and paste it with CTRL/V or some similar on DOS side (e.g. with Shift modifier etc)?

**Comment by stsp** [comment]
I fixed the cursor visibility problems in SDL.
The rest is not a bug, just sdl uses the clipboard.

**Comment by fhanzlik** [comment]
I just test 447.gd806ee3. Behavior now seems be this:
- X mode: in DOS is possible to select text and paste it to DOS or Linux. Is possible to select text in Linux (using primary selection or clipboard) and paste it to DOS with middle mouse button click. When both primary selection and also clipboard are used, middle button click paste to DOS primary selection (is there a possibility in dosemu to paste text in this case from clipboard rather than from primary selection?).

- in SDL mode: IS NOT POSSIBLE select text in DOS. And IS NOT POSSIBLE paste text from Linux (either clipboard or primary selection) to DOS with middle button click.
Cursor visibility is now fixed, it is right.

- in terminal mode (dosemu -t): IS NOT POSSIBLE select text in DOS. And IS NOT POSSIBLE paste text from Linux (either clipboard or primary selection) to DOS with middle button click.

In short, it seems as dosemu in X11 mode behaves perhaps well (without possibility to select between clipboard and primary selection when pasting from Linux to DOS and without possibility to copy in DOS selected text to clipboard).
In SDL and terminal mode text copy/paste isn't working (and it worked before in dosemu 1.4 and also in dosemu2 cca 2 years back /I found the old version commit gc1ea267 from about half of Februar 2015/).

**Comment by stsp** [comment]
No, it works properly, use shift,

**Comment by fhanzlik** [comment]
Yes, You are right, shift did this trick. It work in both SDL and terminal mode. Thanks!

Only one glitch I found: in terminal mode, when selecting text in DOS, I'm able paste it to Linux well. But when pasting it to DOS, only 15 characters are pasted (not all selected text).

**Comment by fhanzlik** [comment]
Same when paste text from Linux to DOS - only 15 chars is pasted.

**Comment by stsp** [comment]
Should now be fixed.

**Comment by fhanzlik** [comment]
I confirm that now work well. Thanks!

**Comment by xucaen** [comment]
How does pressing shift work when pasting text into dosemu? I press shift and noithing happens. I press shift + right click and nothing happens. I press shift+CTRL+C and nothing happens. If paste is working how do I get it to work?

**Comment by stsp** [comment]
How about hold shift and
select some text with a mouse?
